zebrad/src/application.rs

```diff
@@ -393,14 +394,15 @@ impl Application for ZebradApp {
     }
 
     /// Load this application's configuration and initialize its components.
+    #[allow(clippy::unwrap_in_result)]
     fn init(&mut self, command: &Self::Cmd) -> Result<(), FrameworkError> {
         // Create and register components with the application.
         // We do this first to calculate a proper dependency ordering before
         // application configuration is processed
         self.register_components(command)?;
 
         // Fire callback to signal state in the application lifecycle
-        let config = self.config.take().unwrap();
+        let config = self.config.take().expect("we should have a config object");
```

Comment:
We need to explain why the panic never happens.

In this case, we are using `abscissa` in an unusual way, so we should explain exactly what we're doing.

```suggestion
        let config = self.config.take().expect("register_components always populates the config");
```